README.md

```diff
@@ -134,11 +134,18 @@ it to be saved.
 
 **Required option**
 
-This is the secret used to sign the session ID cookie. This can be either a string
-for a single secret, or an array of multiple secrets. If an array of secrets is
+This is the secret used to sign the session ID cookie. This can be either a single 
+secret, or an array of multiple secrets. Secrets can either be a string or a function
+which has `req` passed to it for each request. If an array of secrets is
 provided, only the first element will be used to sign the session ID cookie, while
 all the elements will be considered when verifying the signature in requests.
 
+```js
+app.use(session({
+  secret: ['keyboard cat', function (req) { return 'Danger Zone!' + req.originalUrl }]
```

Comment:
The way secrets work with HMACs, adding something that is user-controlled does not gain anything. Can you provide an example that is more true to a real use-case in the documentation and tests?